float.js

```diff
@@ -1,327 +1,207 @@
 
-    /*
-        Handles DOM communication to this content script
-    */
-    function bindWindowListener() {
-        window.processingfloat = 0;
-
-        window.addEventListener("message", function(event) {
-            if (event.data.host &&
-               (event.data.host === "CSGOFloat") && 
-               event.data.source === "contentscript"){
-                
-                var itemid = event.data.id;
-
-                if (event.data.type == "success") {
-                    // We're no longer processing a float
-                    window.processingfloat = 0;
-
-                    // Add it to a dict holding the requested floats
-                    floatdata[itemid] = event.data.iteminfo;
-
-                    // Display it to the user
-                    processSuccess(itemid);
-                }
-                else if (event.data.type == "error") {
-                    // Display the error to the user
-                    processError(itemid, event.data.error, true);
-                }
-
-                // If this is part of a queue, pop the last element (this request)
-                if (window.floatQueue.length > 0) {
-                    window.floatQueue.pop();
-                }
-
-                // If there are still items in the queue, request the next one
-                if (window.floatQueue.length > 0) {
-                    var queuelength = window.floatQueue.length;
-                    
-                    getCSGOFloat(window.floatQueue[queuelength-1][1], window.floatQueue[queuelength-1][0]);
-                }
-                else {
-                    // There is no longer a queue, show the "Get all floats" button again
-                    $J("#allfloatbutton").show();
-                }
-            }
-        }, false);
-    }
+    if (floatDiv) {
+        // Remove the "get float" button
+        let floatButton = floatDiv.querySelector('.floatbutton');
+        if (floatButton) { floatDiv.removeChild(floatButton); }
 
-    /*
-        Changes the "Get float" button for the specified itemid to the specified message
-    */
-    function changeButtonStatus(itemid, msg) {
-        var floatdiv = $J("#" + itemid + "_floatdiv");
-        if (floatdiv.length > 0) {
-            floatdiv.find("#floatbutton").find("span").text(msg);
-        }
+        let itemInfo = floatData[listingId];
+
+        // Show the float and paint seed to the user
+        let msgdiv = floatDiv.querySelector('.floatmessage');
+        msgdiv.innerHTML = `Float: ${itemInfo.floatvalue}<br>Paint Seed: ${itemInfo.paintseed}`;
     }
+};
 
-    /*
-        Processes a successful float retrieval for the given itemid
-    */
-    function processSuccess(itemid) {
-        // find the corresponding div
-        var floatdiv = $J("#" + itemid + "_floatdiv");
+let processFloatQueue = function() {
+    if (floatQueue.length === 0) { return setTimeout(processFloatQueue, 500); }
```

Comment:
I think the checking interval should be lower than `500` ms, maybe something like `100` to have a more responsive feel.